File: baekjoon/level/04_array1/Python/8958.py
# 8958번 - OX 퀴즈

n = int(input())

for i in range(n):
    l = input()
    sum = 0
    s = 0
    for j in range(len(l)):
        if l[j] == 'O':
            s += 1
            sum += s
        else:
            s = 0
    print(sum)

File: baekjoon/level/04_array1/C/8958.c
// OX퀴즈
#pragma warning(disable:4996)
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(){
    int n, i, score, sum;
    char arr[81], *p=NULL;

    scanf("%d",&n);
    getchar();

    for(i=0; i<n; i++){
        gets(arr);
        score = 0;
        sum = 0;
        for(p=arr;*p!=NULL; p++){
            if(*p=='O'){
                score+=1;
                sum+=score;
            }
            else if(*p=='X'){
                score=0;
            }
        }
        printf("%d\n",sum);
    }

    return 0;
}